src/core/projectinfo.cpp

```diff
@@ -269,21 +271,54 @@ void ProjectInfo::saveLayerTreeState()
   }
 }
 
-void ProjectInfo::saveSnappingConfiguration()
+bool ProjectInfo::snappingEnabled() const
+{
+  if ( mFilePath.isEmpty() )
+    return false;
+
+  return mSettings.value( QStringLiteral( "/qgis/projectInfo/%1/layerSnapping/enabled" ).arg( mFilePath ), false ).toBool();
+}
+
+void ProjectInfo::setSnappingEnabled( bool enabled )
 {
   if ( mFilePath.isEmpty() )
     return;
 
+  mSettings.beginGroup( QStringLiteral( "/qgis/projectInfo/%1/layerSnapping" ).arg( mFilePath ) );
+  mSettings.setValue( QStringLiteral( "enabled" ), enabled );
+  mSettings.endGroup();
+
+  emit snappingEnabledChanged();
+}
+
+void ProjectInfo::saveLayerSnappingConfiguration( QgsMapLayer *layer )
+{
+  if ( mFilePath.isEmpty() || !layer )
```

Comment:
```suggestion
  if ( mFilePath.isEmpty() )
```

Covered by the check below

Reply:
Fair point.